vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -108,25 +108,18 @@ func (r *recommender) UpdateVPAs() {
 		vpa.UpdateConditions(hasMatchingPods)
 		if err := r.clusterState.RecordRecommendation(vpa, time.Now()); err != nil {
 			klog.Warningf("%v", err)
-			if klog.V(4).Enabled() {
```

Comment:
This might have performance implications. Previously this line was gated behind the V(4) meaning the code did never get executed at all under V<4.
I'd prefer keeping the levels unchanged, using V(n).Enabled() often is done to avoid log spam and cost of evaluating log statements when not needed.

Reply:
Understood. Ill revert.